nsxt/resource_nsxt_policy_context_profile.go

```diff
@@ -0,0 +1,479 @@
+/* Copyright © 2019 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	gm_infra "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra"
+	gm_cont_prof "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/global_infra/context_profiles"
+	gm_model "github.com/vmware/vsphere-automation-sdk-go/services/nsxt-gm/model"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	cont_prof "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/context_profiles"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"log"
+)
+
+var attributeDataTypes = []string{
+	model.PolicyAttributes_DATATYPE_STRING,
+}
+
+var attributeKeys = []string{
+	model.PolicyAttributes_KEY_APP_ID,
+	model.PolicyAttributes_KEY_DOMAIN_NAME,
+	model.PolicyAttributes_KEY_URL_CATEGORY,
+}
+
+var subAttributeDataTypes = []string{
+	model.PolicySubAttributes_DATATYPE_STRING,
+}
+
+var subAttributeKeys = []string{
+	model.PolicySubAttributes_KEY_CIFS_SMB_VERSION,
+	model.PolicySubAttributes_KEY_TLS_CIPHER_SUITE,
+	model.PolicySubAttributes_KEY_TLS_VERSION,
+}
+
+func resourceNsxtPolicyContextProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyContextProfileCreate,
+		Read:   resourceNsxtPolicyContextProfileRead,
+		Update: resourceNsxtPolicyContextProfileUpdate,
+		Delete: resourceNsxtPolicyContextProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"attribute":    getContextProfilePolicyAttributesSchema(),
+		},
+	}
+}
+
+func getContextProfilePolicyAttributesSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
```

Comment:
Are we sure order is not important here?

Reply:
Attribute order was not mentioned in the API doc. I did some experiment on UI and did not find the order of attributes  matter when creating context profiles.